UPGRADE_GUIDE.md

```diff
@@ -2,8 +2,133 @@
+
+Previously used `.save()` instance methods are now static `.update()` functions where you specify first the ID of the object you are updating and second, the parameters that need updating.
+
+Functions no longer accept an API key as an optional parameter. If you need per-function API key changes, create a new EasyPostClient object and call the function on the new client that uses the API key you need.
+
+## 5.0 Updated Dependencies
+
+Likelihood of Impact: High
+
+**Ruby 2.6 Required**
+
+easypost-ruby now requires Ruby 2.6 or greater.
+
+**Dependencies**
+
+All dependencies had minor version bumps.
+
+## 5.0 Improved Error Handling
+
+Likelihood of Impact: High
+
+There are ~2 dozen new error types that extend either `ApiError` or `EasyPostError`.
+
+New ApiErrors (extends EasyPostError):
+
+- `ApiError`
+- `ConnectionError`
+- `ExternalApiError`
+- `ForbiddenError`
+- `GatewayTimeoutError`
+- `InternalServerError`
+- `InvalidRequestError`
+- `MethodNotAllowedError`
+- `NotFoundError`
+- `PaymentError`
+- `ProxyError`
+- `RateLimitError`
+- `RedirectError`
+- `RetryError`
+- `ServiceUnavailableError`
+- `SSLError`
+- `TimeoutError`
+- `UnauthorizedError`
+- `UnknownApiError`
+
+New EasyPostErrors (extends builtin Exception):
+
+- `EasyPostError`
+- `EndOfPaginationError`
+- `FilteringError`
+- `InvalidObjectError`
+- `InvalidParameterError`
+- `MissingParameterError`
+- `SignatureVerificationError`
+
+ApiErrors will behave like the previous Error class did. They will include a `message`, `http_status`, and `http_body`. Additionally, a new `code` and `errors` keys are present and populate when available. This class extends the more generic `EasyPostError` which only contains a message, used for errors thrown directly from this library.
+
+## 5.0 Corrected Naming Conventions
+
+Likelihood of Impact: Medium
+
+Occurances of `referral` are now `referral_customer` and `Referral` are now `ReferralCustomer` to match the documentation and API.
+
+Occurances of `smartrate` are now `smart_rate` and `Smartrate` are now `SmartRate` to match the documentation and API.
+
+Occurances of `scanform` are now `scan_form` and `Scanform` are now `ScanForm` to match the documentation and API.
+
+Rename function `get_smartrates` to `get_smart_rates`
+
+Rename function `get_lowest_smartrate` to `get_lowest_smart_rate`
+
+## 5.0 Beta Namespace Changed
+
+Likelihood of Impact: Low
+
+Previously, the beta namespace was found at `easypost.beta.x` where `x` is the name of your model. Now, the beta namespace is simply prepended to the name of your service: `client.beta_x`. for instance, you can call `client.beta_referral_customer.add_payment_method()`.
+
+## 5.0 Return True For Empty API Response
+
+Likelihood of Impact: Low
+
+Empty API response functions for `delete` return `true` instead of empty object
```

Comment:
issue: in Ruby, this didn't change so there is no need for an entry here

Reply:
In V4, the `delete` function will return `self` which is the current object, examples: https://github.com/EasyPost/easypost-ruby/blob/v4/lib/easypost/webhook.rb#L22, https://github.com/EasyPost/easypost-ruby/blob/v4/lib/easypost/user.rb#L26